datatypes/src/collections/feature_collection.rs

```diff
@@ -714,6 +714,45 @@ where
     }
 }
 
+#[derive(Debug)]
+pub struct FeatureCollectionRow<Geometry> {
+    pub geometry: Geometry,
+    pub time_interval: TimeInterval,
+    pub data: HashMap<String, FeatureDataValue>,
```

Comment:
Is it possible to store a reference to `FeatureCollectionIterator` here?
If so, I have two suggestions:
1. Store a GeometryRef instead of the Geometry itself (safe the clone if possible).
2. Remove the hashmap and create a get method that performs the lookup against `data` of `FeatureCollectionIterator`.

For `TimeInterval` it's fine since it`s cheap to copy.